### PR TITLE
Angular14 Upgrade

### PR DESCRIPTION
**Pull request by Rlcolli4**
### Changes
- package.json packages were updated to be Angular 14 or higher versions
- Other dependencies were updated to their latest versions for compatibility 
- Changed version number to 6.0.0
- Added a pending version number and notes to the change log.

### References
- [Issue 712](https://github.com/auth0/angular2-jwt/issues/712)
- [Issue 697](https://github.com/auth0/angular2-jwt/issues/697)

### Checklist
- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All code quality tools/guidelines in the [CONTRIBUTING documentation](https://github.com/auth0/open-source-template/blob/master/CONTRIBUTING.md) have been run/followed
- [X] All relevant assets have been compiled as directed in the [CONTRIBUTING documentation](https://github.com/auth0/open-source-template/blob/master/CONTRIBUTING.md), if applicable

### CODE REVIEW

File: package.json
Comment (demkalkov) [inline package.json:45]: Eslint should be used now instead of tslint and codelyzer

File: angular.json
Comment (demkalkov) [inline angular.json:100]: Why did you remove lint?

**Review by demkalkov — CHANGES_REQUESTED** [review]
Please replace tslint with eslint

**Comment by Rlcolli4** [comment]
I'll fix this as soon as I can. I didn't remove it intentionally, I forked the master branch and did the upgrade. It's probably something I have locally that did it without my knowledge.

**Comment by Rlcolli4** [comment]
Just migrated from tslint/codelyzer to eslint. Added lint back to places in the angular.json based on steps and documentation for converting.

**Comment by MumiaIrrequieta** [comment]
Can we ask @Sambego to review and merge this? "auth0/developer-relations-admin" link is dead so maybe no one is receiving the notifications and he's the latest user to commit to this project.
Thanks.

**Comment by juvegitau** [comment]
@Sambego are in a position to merge?

File: CHANGELOG.md
Comment (frederikprijck) [inline CHANGELOG.md:5]: Can you revert these changes please? We will update the changelog when we cut a release.

File: package.json
Comment (frederikprijck) [inline package.json:3]: Can you revert the version change please?

```suggestion
  "version": "5.0.2",
```

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Thanks for the PR. As mentioned in #712 , apologies about the radio silence here.

Could you also update the package-lock.json file to remove all the entries to `npmjs.sitesusa.local`?

File: angular.json
Comment (frederikprijck) [inline angular.json:110]: Why does the linter uses the e2e tsconfig?

File: angular.json
Comment (frederikprijck) [inline angular.json:108]: This is failing. We should update the builder to use `"@angular-eslint/builder:lint"`

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Comment by frederikprijck** [comment]
Doing a fresh pull from this branch, removing the lock file followed by `npm i` results in an error.

The following changes should solve that, can you update the PR to incorporate these as well?

- Set `jasmine-core` to `~3.8.0`
- Set `karma-jasmine-html-reporter` to `^1.7.0`

Thanks!

**Comment by Rlcolli4** [comment]
I updated the package.json and angular.json as requested! The two latest commits handled the reversion and then any other changes I needed to get `npm start`, `npm run build`, `npm run e2e` (which does appear to be out of date, its looking for text that I couldn't find anywhere in the app), and `npm run test` to work.

File: angular.json
Comment (frederikprijck) [inline angular.json:112]: This is linting the wrong files, it should lint inside `src` for this one.

File: angular.json
Comment (frederikprijck) [inline angular.json:171]: I dont think this should have a `tsConfig`; it should also include the files from `project/angular-jwt`, basically the lint config above should be here, and the config above should lint `src`.

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Comment by frederikprijck** [comment]
Running `npm run lint` throws an exception, can you look into this?

Btw, I know this PR is abit old, and I don't expect you to keep putting effort here to get this merged. Let me know if you want me to patch it as needed for you. That said, more than happy to leave it to you, just feel free to let me know!

**Comment by Rlcolli4** [comment]
@frederikprijck I'll take a look at these this afternoon. I don't mind continuing to improve this PR if it helps your team focused on getting settled with the project. I am on the west coast of the US so I apologize if these responses appear as being "late".

**Review by frederikprijck — APPROVED** [review]

**Comment by frederikprijck** [comment]
Thanks for the PR 🚀 . Can you resolve the conflict and then we should be good to merge.

**Comment by Rlcolli4** [comment]
Merge is done

**Comment by frederikprijck** [comment]
First of all, thanks for the PR. Really appreciate the effort here, as this has been a PR of high quality.

I had a conversation with the Angular team and concluded the following:

- When you compile an SDK with Angular 14 and compilationMode set to `partial`, the SDK will only work with Angular 14+.

However, as we still need to support Angular 12 and 13, it appears to be that we have two options:

- Create a separate version for every major version of Angular
- Compile using the lowest supported Angular version, currently being 12.

Given we have no specific features from angular apart from a service and an interceptor, I believe we should be fine to stick with Angular 12.

However, that would mean we can not merge this PR yet, but can only do so once Angular has dropped support for Angular 12 and 13.

I feel a bit bad about that, because of the fact that this PR has been open for so long, and all I wanted was get some activity in here and progress the repository as well as get your work in. But I believe we should not merge this PR because of what is mentioned above. I hope you understand the reasoning, and my sincere apologies.

That said, I will bump everything to Angular 12, compile using partial mode and get rid of the legacy warning when compiling.

**Comment by frederikprijck** [comment]
Closing in favor of #735 and #734

**Pull request closed without merging**
